src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -194,22 +191,36 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
             {
                 // We are on the wrong thread.
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer did not provide Edit Access"));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
             if (subjectBuffer.EditInProgress)
             {
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer is editing by someone else."));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
+            CompletionChange change;
+
+            // We met an issue when external code threw an OperationCanceledException and the cancellationToken is not cancelled.
+            // Catching this scenario for further investigations.
```

Comment:
Can this link to whatever bug we're trying to investigate? Totally seen times where we found the underlying bug and then forgot to delete the reporting code, and it's really hard to know what the issue was looking for...